spec/pr_size_checker_spec.rb

```diff
@@ -0,0 +1,194 @@
+          allow(@plugin.git).to receive(:modified_files).and_return([])
+          allow(@plugin.git).to receive(:deleted_files).and_return([])
+        end
+
+        shared_examples 'using the default diff size counter without a file selector' do |type|
+          let(:diff_counter_for_type) do
+            type_hash = {
+              insertions: :insertions,
+              deletions: :deletions,
+              all: :lines_of_code
+            }
+
+            type_hash[type]
+          end
+
+          it 'returns a warning when using default parameters in a PR that has larger diff than the default maximum' do
+            allow(@plugin.git).to receive(diff_counter_for_type).and_return(501)
+
+            @plugin.check_diff_size(type: type)
+
+            expect(@dangerfile.status_report[:warnings]).to eq ['Please keep the Pull Request small, breaking it down into multiple ones if necessary']
+          end
+
+          it 'does nothing when using default parameters in a PR that has equal diff than the default maximum' do
+            allow(@plugin.git).to receive(diff_counter_for_type).and_return(500)
+
+            @plugin.check_diff_size(type: type)
+
+            expect(@dangerfile.status_report[:errors]).to be_empty
+            expect(@dangerfile.status_report[:warnings]).to be_empty
+          end
+
+          it 'does nothing when using default parameters in a PR that has smaller diff than the default maximum' do
+            allow(@plugin.git).to receive(diff_counter_for_type).and_return(499)
+
+            @plugin.check_diff_size(type: type)
+
+            expect(@dangerfile.status_report[:errors]).to be_empty
+            expect(@dangerfile.status_report[:warnings]).to be_empty
+          end
+
+          it 'returns an error using a custom message and a custom pr body size' do
+            allow(@plugin.git).to receive(diff_counter_for_type).and_return(600)
+
+            custom_message = 'diff size custom message'
+            @plugin.check_diff_size(type: type, max_size: 599, message: custom_message, fail_on_error: true)
+
+            expect(@dangerfile.status_report[:errors]).to eq [custom_message]
+          end
+        end
+
+        shared_examples 'using a file selector to filter and count the changes in a diff' do |type, max_sizes|
+          it 'returns a warning when using a files filter that will result in a diff that is too large' do
+            prepare_diff_with_test_files
+
+            @plugin.check_diff_size(
+              file_selector: ->(path) { File.dirname(path).start_with?('src/test/java') },
+              type: type,
+              max_size: max_sizes[0]
+            )
+
+            expect(@dangerfile.status_report[:warnings]).to eq ['Please keep the Pull Request small, breaking it down into multiple ones if necessary']
+          end
+
+          it 'returns a warning when using a files filter that will result in a diff that is too large, with a custom error' do
```

Comment:
Have you considered using a `context` here?

```ruby
context 'when using a files filter that will regard the diff as too large' do
  it 'reports a warning' do ...
  it 'reports a warning with a given custom message' do ...
 ...
```

Reply:
When I read this suggestion I wondered if it wouldn't be an overkill (it's a `shared_examples` with 3 examples); but I tried it out and liked how it broke down the statements, it is more readable now. Thanks for the suggestion!